http-body-util/src/combinators/map_data.rs

```diff
@@ -56,26 +56,26 @@ where
     type Data = B2;
     type Error = B::Error;
 
-    fn poll_data(
+    fn poll_frame(
         self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-    ) -> Poll<Option<Result<Self::Data, Self::Error>>> {
+    ) -> Poll<Option<Result<Frame<Self::Data>, Self::Error>>> {
         let this = self.project();
-        match this.inner.poll_data(cx) {
+        match this.inner.poll_frame(cx) {
             Poll::Pending => Poll::Pending,
             Poll::Ready(None) => Poll::Ready(None),
-            Poll::Ready(Some(Ok(data))) => Poll::Ready(Some(Ok((this.f)(data)))),
+            Poll::Ready(Some(Ok(frame))) if frame.is_data() => {
+                Poll::Ready(Some(Ok(Frame::data((this.f)(frame.into_data())))))
+            }
+            Poll::Ready(Some(Ok(_other))) => {
```

Comment:
Would it make sense to make this mapper more generic? So make this a `MapFrame` instead of data and then the users closure can figure out what to do per frame etc.

Reply:
We could. Mapping a frame is probably useful. I think mapping DATA is the _most_ useful, and so probably worth being a dedicated combinator.